llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtInterfaces.td

```diff
@@ -40,6 +40,16 @@ def VectorLayoutInterface : AttrInterface<"VectorLayoutInterface"> {
       /*retTy=*/"SmallVector<int64_t>",
       /*methodName=*/"getDistributedShape",
       /*args=*/(ins)
+    >,
+    InterfaceMethod<
+      /*description=*/"Get the delinearized ids of the given thread id.",
+      /*retTy=*/"FailureOr<ValueRange>",
+      /*methodName=*/"computeThreadIds",
+      /*args=*/(ins "Value":$threadId, "RewriterBase&":$rewriter),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{
+        return failure();
+      }]
```

Comment:
This method doesn't really belong on this interface. VectorLayoutInterface should not know about anything GPU-related. I was hoping for a "GPUVectorLayoutInterface" exposing this interface. It's fine if we want to refactor this later, but a TODO and an issue for this would be good.

Reply:
I can move it to a method on the nested layout attribute itself.